src/agentscope/parsers/json_object_parser.py

```diff
@@ -195,8 +209,23 @@ def __init__(
                 - `True`, the whole dictionary will be returned
 
         """
-        # Initialize the markdown json object parser
-        MarkdownJsonObjectParser.__init__(self, content_hint)
+        self.pydantic_class = None
+
+        # Initialize the content_hint according to the type of content_hint
+        if inspect.isclass(content_hint) and issubclass(
+            content_hint,
+            BaseModel,
+        ):
+            self.pydantic_class = content_hint
+            self.content_hint = "{a_JSON_dictionary}"
```

Comment:
1. Yes, validation with pydantic is much more challenging for LLMs. 
2. I have considered to combine available messages into an example dictionary in fenced code block, but it adds additional requirements for the `Field()` object, e.g. the description parameters must be set.

Reply:
I agree that the goal is to ensure what _types_ should return rather than what _correct content_ should return. So the key snippet is `response.parsed = dict(self.pydantic_class(**response.parsed))` rather than `self.format_instruction`

At the same time, developers should be reminded in the tutorial that they can subclass `MarkdownJsonDictParser` and re-implement `self.format_instruction` if they are not satifised with the prompt

Reply:
> I agree that the goal is to ensure what _correct types_ should return rather than what _correct content_ should return. So the key snippet is `response.parsed = dict(self.pydantic_class(**response.parsed))` rather than `self.format_instruction`
> 
> At the same time, developers should be reminded in the tutorial that they can subclass `MarkdownJsonDictParser` and re-implement `self.format_instruction` if they are not satifised with the prompt

Tutorial updated.